### ISSUE TITLE
Compiler API compiles arrow functions without curly brackets

### ISSUE THREAD

**Issue by life777** [Microsoft/TypeScript#9326]
Hi,

I use typescript compiler API for compiling files from [this article](https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API). But instead of setting target in compile options like ts.ScriptTarget.ES5, I read from file system a tsconfig.json file and pass "compilerOptions" from it to the ts.createProgram as a second parameter. I ran into some problem with compiling. If I pass to the ts.createProgram options with invalid target value, it will compile arrow functions without curly brackets.

**TypeScript Version:** 

1.8.10

**Code**

```js
var ts = require("./tsCompiler/typescript"),
      program = ts.createProgram(["./program.ts"], { target: "ES5", module: "CommonJS"}),
      emitResult = program.emit();
```

program.ts looks like:

```ts
var b = [1,2,3].filter(a => a < 2);
```

**Expected behavior:**

I expected two possible results:
1) It should throw an error because of invalid target value in compiler options.
2) Instead of invalid compile options it will use a default "ES3" target and compile the code of program.ts to:

```js
var b = [1,2,3].filter(function(a) { return a < 2; });
``` 

**Actual behavior:** 

It compiles without errors and generate invalid javascript code without curly brackets like:

```js
var b = [1, 2, 3].filter(function (a) a < 2);
```

Thanks in advance.

**Comment by vladima** [comment]
looks like a dupe of https://github.com/Microsoft/TypeScript/issues/7980 that should be fixed in master. Can you give a try to `typescript@next` on npm to see if you still see this issue?

**Comment by life777** [comment]
Yes, it still exists. I use "ES5" and "es5" string intead of ts.ScriptTarget.ES5.

**Comment by mhegazy** [comment]
The `createProgram` API does not accept strings.  please see the declaration in https://github.com/Microsoft/TypeScript/blob/c98166ccc7b92e41d09e640fb4f06ac35385d49d/lib/typescriptServices.d.ts#L1852. you need to pass it a valid [`CompilerOptions`](https://github.com/Microsoft/TypeScript/blob/c98166ccc7b92e41d09e640fb4f06ac35385d49d/lib/typescriptServices.d.ts#L1524), and that expects a numeric value that is part of the ScriptTarget enum and not a string.

Also it is worth noting if you were to compile this code with the TS compiler, you woudl get:

```txt
error TS2345: Argument of type '{ target: string; module: string; }' is not assignable to parameter of type 'CompilerOptions'.
  Types of property 'module' are incompatible.
    Type 'string' is not assignable to type 'ModuleKind'.
```
